src/Core/Services/StateMigrationService.cs

```diff
@@ -18,6 +19,8 @@ public class StateMigrationService : IStateMigrationService
         private readonly IStorageService _liteDbStorageService;
         private readonly IStorageService _secureStorageService;
 
+        private static SemaphoreSlim _semaphore;
```

Comment:
This service Is registered and used as a Singleton. Do you need the `static` keyword because of multiple threads accessing it or something like that? Because I think it shouldn't be necessary.
Btw, if it needs to be `static` then you can directly initialize it inline here:
```c#
private static SemaphoreSlim _semaphore = new SemaphoreSlim(1);
```

Reply:
You are correct, I overlooked the singleton nature of the service.